solidity/contracts/isms/hook/AbstractMessageIdAuthorizedIsm.sol

```diff
@@ -72,17 +72,23 @@ abstract contract AbstractMessageIdAuthorizedIsm is
      */
     function verify(
         bytes calldata,
-        /*_metadata*/
-        bytes calldata message
-    ) external returns (bool) {
-        bytes32 messageId = message.id();
+        /*metadata*/ bytes calldata message
+    ) external virtual returns (bool) {
+        return releaseValueToRecipient(message);
+    }
 
-        // check for the first bit (used for verification)
-        bool verified = verifiedMessages[messageId].isBitSet(
-            VERIFIED_MASK_INDEX
-        );
-        // rest 255 bits contains the msg.value passed from the hook
+    // ============ Public Functions ============
+
+    /**
+     * @notice Release the value to the recipient if the message is verified.
+     * @param message Message to release value for.
+     */
+    function releaseValueToRecipient(
+        bytes calldata message
+    ) public returns (bool) {
+        bool verified = isVerified(message);
```

Comment:
I dont think this function should be responsible for verifying
it might be slightly more readable if you move this `verified` branch to the `verify` function?
you are also reading from the same storage slot twice here

Reply:
fixed